nest-forms-backend/src/config/ba-config.instance.ts

```diff
@@ -0,0 +1,13 @@
+import BaConfig from './ba-config'
+import validateEnvironmentVariables from './environment-variables.validate'
+
+let instance: BaConfig | null = null
+
+export default function getBaConfigInstance(): BaConfig {
+  if (!instance) {
+    const validatedEnvVariables = validateEnvironmentVariables(process.env)
+    instance = new BaConfig(validatedEnvVariables)
+  }
```

Comment:
I understand, that this all will be initialized in a single thread, so this implementation should be sufficient. However, if it is not, it would need a way to synchronize in order not to possibly create a phantom instances (after multiple threads pass the `!instance` check before any assigns into the `instance` variable).